doc/source/train/user-guides/experiment-tracking.rst

```diff
@@ -1,195 +1,292 @@
-you can use the regular tracking library APIs to report your results, models, and
-other artifacts.
+.. note::
+    This guide is relevant for all DataParallelTrainer, including TorchTrainer and TensorflowTrainer.
 
-Ray Train also exposes logging callbacks that automate some of these tasks.
+Most experiment tracking libraries work out-of-the-box with Ray Train. 
+In the following guide, you will learn how to set up code so that your favorite experiment tracking libraries 
+can work with distributed data parallel training with Ray Train. We will conclude the session with debugging
+tips.
 
-.. _train-experiment-tracking-native:
+Ray Train lets you use native experiment tracking libraries by customizing your tracking 
+logic inside ``train_loop_per_worker``. 
+This gives you the highest degree of freedom to decide when and where to log params, metrics, 
+models or even media contents. 
+This also allows you to use some native integrations that these tracking frameworks have with 
+specific training frameworks, for example ``mlflow.pytorch.autolog()``, 
+``lightning.pytorch.loggers.MLFlowLogger`` etc. 
+In this way, you get to port your experiment tracking logic to Ray Train with minimal change. 
 
-Using native experiment tracking libraries
-------------------------------------------
+How to set up your code to log distributed training experiment
+==============================================================
 
-.. tab-set::
+Step 1: Decide a tracking service to use
+----------------------------------------
 
-    .. tab-item:: PyTorch
+For easy configuration, we recommend the following:
 
-        You can use experiment tracking libraries such as Weights & Biases, Mlflow, or
-        Comet directly in your Ray Train training loop.
+**MLflow:**
 
-        There are two things to keep in mind:
+- offline mode: file based tracking uri pointing to shared storage path (``mlflow.start_run(tracking_uri=”file:some_shared_storage_path”)``)
+- online mode: externally hosted by Databricks(``mlflow.start_run(tracking_uri=”databricks”)``).
 
-        1. Your code is executed in parallel on many workers. However, you often only want to report
-        results from one of these workers (usually the first worker - the "rank 0" worker).
+**TensorBoard:**
 
-        2. When using the native libraries, you should report the results to both Ray Train and
-        the experiment tracking library.
+A shared storage location where all nodes can write to.
 
-        Example:
+**W&B:**
 
-        .. code-block:: python
+- offline mode: directory pointing to shared storage path (``wandb.init(dir="some_shared_storage_path")``)
+- online mode: should work out-of-the-box with correct credentials
+
+Step 2: Make sure that all nodes and processes have access to credentials
+-------------------------------------------------------------------------
+
+See example below.
+
+Step3: Initialize the run 
```

Comment:
Can we add some simple code snippets for each step to make it clearer?

Reply:
I feel like I am repeating myself. The code snippet is shown down below. But I can touch on it a bit.